src/sentry/tasks/auto_archive_issues.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from datetime import datetime, timedelta
+from typing import List
+
+import pytz
+from sentry_sdk.crons.decorator import monitor
+
+from sentry import features
+from sentry.models import (
+    Activity,
+    Group,
+    GroupHistory,
+    GroupHistoryStatus,
+    GroupStatus,
+    ObjectStatus,
+    Organization,
+    OrganizationStatus,
+    Project,
+    record_group_history_from_activity_type,
+    remove_group_from_inbox,
+)
+from sentry.tasks.base import instrumented_task
+from sentry.types.activity import ActivityType
+from sentry.types.group import GroupSubStatus
+from sentry.utils.iterators import chunked
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = logging.getLogger(__name__)
+
+ITERATOR_CHUNK = 10_000
+
+
+@instrumented_task(
+    name="sentry.tasks.auto_archive_issues.run_auto_archive",
+    queue="auto_transition_issue_states",
+    max_retries=3,
+)  # type: ignore
+@monitor(monitor_slug="auto-archive-job-monitor")
+def run_auto_archive() -> None:
+    """
+    Automatically transition issues that are ongoing for 14 days to archived until escalating.
+    """
+    logger.info("Starting task for sentry.tasks.weekly_escalating_forecast.run_escalating_forecast")
+
+    for organization in RangeQuerySetWrapper(
+        Organization.objects.filter(status=OrganizationStatus.ACTIVE)
+    ):
+        if features.has("organizations:escalating-issues", organization):
+            project_ids = list(
+                Project.objects.filter(
+                    organization_id=organization.id, status=ObjectStatus.ACTIVE
+                ).values_list("id", flat=True)
+            )
+            run_auto_achive_for_projects.delay(project_ids=project_ids)
+
+
+@instrumented_task(
+    name="sentry.tasks.auto_archive_issues.run_auto_archive_for_projects",
+    queue="auto_transition_issue_states",
+    max_retries=3,
+    default_retry_delay=60,
+)  # type: ignore
+def run_auto_achive_for_projects(project_ids: List[int]) -> None:
+    now = datetime.now(tz=pytz.UTC)
+    fourteen_days_ago = now - timedelta(days=14)
+
+    for ongoing_groups in chunked(
+        RangeQuerySetWrapper(
+            Group.objects.filter(
+                status=GroupStatus.UNRESOLVED,
+                substatus=GroupSubStatus.ONGOING,
+                project_id__in=project_ids,
+                last_seen__gte=datetime.now() - timedelta(days=14),
+            ),
+            step=ITERATOR_CHUNK,
+        ),
+        ITERATOR_CHUNK,
+    ):
```

Comment:
I think following the pattern we have for auto transitioning regressed issues to ongoing might make sense.